### ISSUE TITLE
Files failing to download - but reported as successful

### ISSUE THREAD

**Issue by gregbert42** [boredazfcuk/docker-icloudpd#8]
Noticing the directories are getting created with 'root' ownership and then photos fail to download. At the end, all is reported as successful!

2020-02-14 22:04:49 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1684.JPG
2020-02-14 22:22:32 ERROR    IOError while writing file to /home/username/iCloud/2020/02/13/IMG_1684.JPG! You might have run out of disk space, or the file might be too large for your OS. Skipping this file...
2020-02-14 22:22:32 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1683.JPG
2020-02-14 22:40:17 ERROR    IOError while writing file to /home/username/iCloud/2020/02/13/IMG_1683.JPG! You might have run out of disk space, or the file might be too large for your OS. Skipping this file...
2020-02-14 22:40:17 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1682.JPG
2020-02-14 22:58:02 ERROR    IOError while writing file to /home/username/iCloud/2020/02/13/IMG_1682.JPG! You might have run out of disk space, or the file might be too large for your OS. Skipping this file...
2020-02-14 22:58:02 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1681.JPG
2020-02-14 23:15:46 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-14 23:33:36 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-14 23:51:26 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-15 00:09:16 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-15 00:27:07 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-15 00:27:12 INFO     Could not download IMG_1681.JPG! Please try again later.
2020-02-15 00:27:12 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1680.JPG
2020-02-15 00:44:57 ERROR    Error downloading IMG_1680.JPG, retrying after 5 seconds...
2020-02-15 01:02:47 ERROR    Error downloading IMG_1680.JPG, retrying after 5 seconds...
2020-02-15 01:20:37 ERROR    Error downloading IMG_1680.JPG, retrying after 5 seconds...
2020-02-14 22:04:42 DEBUG    Looking up all photos and videos from album All Photos...
2020-02-14 22:04:42 INFO     Downloading 3316 original photos and videos to /home/username/iCloud/ ...
2020-02-14 22:04:49 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1684.JPG
2020-02-14 22:22:32 ERROR    IOError while writing file to /home/username/iCloud/2020/02/13/IMG_1684.JPG! You might have run out of disk space, or the file might be too large for your OS. Skipping this file...
2020-02-14 22:22:32 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1683.JPG
2020-02-14 22:40:17 ERROR    IOError while writing file to /home/username/iCloud/2020/02/13/IMG_1683.JPG! You might have run out of disk space, or the file might be too large for your OS. Skipping this file...
2020-02-14 22:40:17 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1682.JPG
2020-02-14 22:58:02 ERROR    IOError while writing file to /home/username/iCloud/2020/02/13/IMG_1682.JPG! You might have run out of disk space, or the file might be too large for your OS. Skipping this file...
2020-02-14 22:58:02 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1681.JPG
2020-02-14 23:15:46 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-14 23:33:36 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-14 23:51:26 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-15 00:09:16 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-15 00:27:07 ERROR    Error downloading IMG_1681.JPG, retrying after 5 seconds...
2020-02-15 00:27:12 INFO     Could not download IMG_1681.JPG! Please try again later.
2020-02-15 00:27:12 INFO     Downloading /home/username/iCloud/2020/02/13/IMG_1680.JPG
2020-02-15 00:44:57 ERROR    Error downloading IMG_1680.JPG, retrying after 5 seconds...
2020-02-15 01:02:47 ERROR    Error downloading IMG_1680.JPG, retrying after 5 seconds...
2020-02-15 01:20:37 ERROR    Error downloading IMG_1680.JPG, retrying after 5 seconds...
(....)
2020-02-15 10:18:28 INFO     All photos have been downloaded!
2020-02-15 10:18:29 INFO     Web cookie expires: 2020-02-28 @ 14:37:14
2020-02-15 10:18:29 INFO     Two factor authentication cookie expires: 2020-03-29 @ 14:37:37

**Comment by boredazfcuk** [comment]
Sorry, had a boozy Saturday night so motivation levels were low yesterday lol

I'm guessing this is due to the bit where it checks for new downloads. I run /usr/bin/icloudpd as root at that point, because I didn't think it mattered. It's only supposed to be printing a list of files and not making any changes... or at least that's what the  "--only-print-filenames" command line argument suggests to me.

I'm guessing it's actually creating the folder structure during this check, so when it comes to downloading the files in the following run, it can't save anything into the folders because they're owned by root and not your ${user} account.

I've changed it so the check now runs as ${user} too. This should mean new folders are created as ${user} so the download run doesn't run into the ownership issue.

I've pushed all container changes... It's prob gonna take DockerHub a while to process everything again as plenty of containers have had updates again...

Any problems, let me know.
